### DISCUSSION TITLE
Same IP for multiple instances

### DISCUSSION THREAD

**Discussion by junnplus** [lima-vm/lima#690] [Q&A] [answered]
### Description

```
[N] ❯ limactl list
NAME       STATUS     SSH                ARCH       CPUS    MEMORY    DISK      DIR
vm1        Running    127.0.0.1:51426    aarch64    4       4GiB      100GiB    /Users/jun/.lima/vm1
vm2        Running    127.0.0.1:51487    aarch64    4       4GiB      100GiB    /Users/jun/.lima/vm2

[N] ❯ limactl shell vm1 ip route
default via 192.168.5.2 dev eth0 proto dhcp src 192.168.5.15 metric 100
192.168.5.0/24 dev eth0 proto kernel scope link src 192.168.5.15
192.168.5.2 dev eth0 proto dhcp scope link src 192.168.5.15 metric 100

[N] ❯ limactl shell vm2 ip route
default via 192.168.5.2 dev eth0 proto dhcp src 192.168.5.15 metric 100
192.168.5.0/24 dev eth0 proto kernel scope link src 192.168.5.15
192.168.5.2 dev eth0 proto dhcp scope link src 192.168.5.15 metric 100
```

Is there a way for them to form a network?

**Comment by afbjorklund** [comment, marked as answer]
It is actually the same IP for _every_ instance. The "user" network is not seen by the others.

https://wiki.qemu.org/Documentation/Networking

The current alternative is VDE, but that is currently _only_ available on macOS (`vde_vmnet`)

It would be possible to set up a "socket" network, that would give them simple connectivity.

----

I will add a separate issue about it, but basically the first VM "listens" to a port on the host.

And then the second (and third...) VM "connects" to this port, so they each get an `eth1`...

Setting up tun/tap (or vde) also for Linux would be more powerful, but also requires root.

* https://github.com/lima-vm/lima/issues/682

**Comment by afbjorklund** [comment]
This was the story about adding Linux networking:

* https://github.com/lima-vm/lima/issues/358
